R/var_label.R

```diff
@@ -224,8 +249,40 @@ set_variable_labels <- function(.data, ..., .labels = NA, .strict = TRUE) {
     }
 
     for (v in intersect(names(values), names(.data)))
-      var_label(.data[[v]]) <- values[[v]]
+      label_attribute(.data[[v]]) <- values[[v]]
   }
 
   .data
 }
+
+#' @rdname var_label
+#' @export
+label_attribute <- function(x) {
+  attr(x, "label", exact = TRUE)
+}
+
+#' @rdname var_label
+#' @export
+`label_attribute<-` <- function(x, value) {
+  if ((!is.character(value) && !is.null(value)) || length(value) > 1)
+    stop(
+      "`value` should be a single character string or NULL",
+      call. = FALSE,
+      domain = "R-labelled"
+    )
+  attr(x, "label") <- value
+  x
+}
+
+#' @rdname var_label
+#' @export
+get_label_attribute <- function(x) {
+  label_attribute(x)
+}
+
+#' @rdname var_label
+#' @export
+set_label_attribute <- function(x, value) {
+  label_attribute(x) <- value
+  x
```

Comment:
I would expect the `<-()` function to call the setter. Besides being shorter, this *might* avoid extra duplication of R objects due to the way how `<-()` works.

Reply:
I'm not sure to understand your point here. 

`set_label_attribute()`, as the other `set_*()` functions, is not supposed to changed the object, but rather to return the changed object.